### ISSUE TITLE
Prebuild N-API module using cmake-js

### ISSUE THREAD

**Issue by s1hofmann** [prebuild/prebuild#253]
**prebuild version**: 9.0.1
**cmake-js version**: 5.3.1
**node version**: 10.16.0
**platform**: macOS 10.14.6

Hi all,

I'm trying to prebuild an N-API module using cmake-js.
Following the readme, my configuration looks like this:

> package.json
```
...
"scripts": {
...
    "install": "prebuild-install --verbose || cmake-js rebuild",
    "prebuild": "prebuild --backend cmake-js -t 3 -r napi",
...
},
"binary": {
    "napi_versions": [
        3,
        4
    ]
}
...
```

`cmake-js rebuild` works fine, but when trying to prebuild, `cmake-js` runs into problems.
```
info TOOL Using Unix Makefiles generator.
ERR! OMG Unknown runtime: napi
ERR! OMG Unknown runtime: napi
```

`prebuild` tries to run `cmake-js` with the following parameters:

```
cmake-js rebuild --runtime-version=3 --arch=x64 --runtime=napi
```

Problem is, `cmake-js` does not know about runtime `napi` or runtime-version `3`:

```
Available settings:

runtime: application's target runtime, possible values are:
    node: Node.js
    nw: nw.js
    electron: Electron
runtimeVersion: version of the application's target runtime, for example: 0.12.1
arch: architecture of application's target runtime (eg: x64, ia32, arm). Notice: on non-Windows systems the C++ toolset's architecture's gonna be used despite this setting. If you don't specify this on Windows, then architecture of the main node/io.js runtime is gonna be used, so you have to choose a matching nw.js runtime.
```

Am I missing out on something? Any help or hints appreciated! 😃 

P.S. Thanks for your work here! We're relying on `prebuild` and `prebuild-install` in our projects and it helps us a lot! Keep up the great work! 👍

**Comment by vweevers** [comment]
Building as a regular node addon (`-r node`) might work. I don't know how `cmake-js` handles N-API versioning if at all.

@oleavr @jschlight @NickNaso Any advice as to what needs to happen here? Thanks!

**Comment by jschlight** [comment]
@s1hofmann is your project publicly available?

**Comment by jschlight** [comment]
It looks like `cmakebuild.js` needs to be updated to support N-API like `gypbuild.js`. I'm reestablishing my development environment and will continue to assess this.

**Comment by s1hofmann** [comment]
@jschlight Unfortunately, the project is not public. Glad to hear that there might be a way to use cmake-js with prebuild and N-API.

Thanks for your effort!

**Comment by jschlight** [comment]
I've got a PR just about ready to submit. All eight Travis CI jobs are successfully running. Two of eight AppVeyor Windows jobs are failing. I'll look more closely at the failed jobs when I return next week from a break.

**Comment by vweevers** [comment]
Available in 9.1.0. See the README or #255 for instructions. Thanks @jschlight!
